src/js/components/PodViewFilter.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+
+import FilterBar from './FilterBar';
+import FilterButtons from './FilterButtons';
+import FilterInputText from './FilterInputText';
+import StringUtil from '../utils/StringUtil';
+
+const METHODS_TO_BIND = [
+  'getFilterButtonContent',
+  'handleStatusFilterChange',
+  'handleSearchStringChange'
+];
+
+class PodViewFilter extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    METHODS_TO_BIND.forEach(function (method) {
+      this[method] = this[method].bind(this);
+    }, this);
+  }
+
+  handleStatusFilterChange(filterByStatus) {
+    this.props.onFilterChange(Object.assign({}, this.props.filter, {
+      status: filterByStatus
+    }));
+  }
+
+  handleSearchStringChange(searchString = '') {
+    this.props.onFilterChange(Object.assign({}, this.props.filter, {
+      text: searchString
+    }));
+  }
+
+  getFilterButtonContent(filterName, count) {
+    return (
+      <span className="button-align-content">
+        <span className="label">{StringUtil.capitalize(filterName)}</span>
+        <span className="badge">{count || 0}</span>
+      </span>
+    );
+  }
+
+  getChildrenArray() {
```

Comment:
I removed this here: https://github.com/dcos/dcos-ui/pull/1069/commits/6254e5cad83be310776401bfb8beeff606b512a2